src/utils/splits.ts

```diff
@@ -90,3 +97,82 @@ export const formatOutgoingSplits = (splits: OutgoingSplit[]): Split[] => {
     ) ?? []
   )
 }
+
+const splitIsProject = (split: Split) =>
+  Boolean(split.projectId && parseInt(split.projectId) > 0)
+
+// determines if two splits are the same 'entity' (either projectId or address)
+const splitsAreSameEntity = (a: Split, b: Split) => {
+  const isProject = splitIsProject(a)
+  const idsEqual =
+    a.projectId === b.projectId ||
+    BigNumber.from(a.projectId).eq(b.projectId ?? 0) ||
+    BigNumber.from(b.projectId).eq(a.projectId ?? 0)
+
+  return (
+    (isProject && idsEqual) || (!isProject && a.beneficiary === b.beneficiary)
+  )
+}
+
+// returns a sorted list of a give list of splits (by percent allocation)
+const sortSplits = (splits: Split[]) => {
+  return splits.sort((a, b) => (a.percent < b.percent ? 1 : -1))
+}
+
+// return list of splits that exist in oldSplits but not newSplits
+export const getRemovedSplits = (oldSplits: Split[], newSplits: Split[]) => {
+  return oldSplits.filter(oldSplit => {
+    return !newSplits.some(newSplit => splitsAreSameEntity(oldSplit, newSplit))
+  })
+}
+
+// returns all unique splits (projectIds or addresses), sorted by their new `percent`
+// and assigns each split an additional property `oldSplit` (OldSplit)
+export const processUniqueSplits = ({
```

Comment:
So all in all, I think we might not need this function as it stands. see https://github.com/jbx-protocol/juice-interface/pull/2710/files#r1062057840